subprojects/logging/src/main/java/org/gradle/internal/logging/LoggingOutputInternal.java

```diff
@@ -46,7 +46,7 @@
      *
      * <p>Removes standard output and/or error as a side-effect.
      */
-    void attachAnsiConsole(OutputStream outputStream);
+    void attachAnsiConsole(OutputStream outputStream, boolean verbose);
```

Comment:
Unless otherwise discussed, the verbose flag should be attached to all outputs as opposed to a specific output. I would probably look into handling the verbose flag the same way as the log level is handled. PR https://github.com/gradle/gradle/pull/2164 was the first try at making the `--dry-run` flag work with the rich console. Maybe you can find something useful for this PR.